### ISSUE TITLE
Auto-attempt to renew IP address

### ISSUE THREAD

**Issue by felipecrs** [make-all/tuya-local#2587]
### Description

Initial report [here](https://github.com/make-all/tuya-local/issues/1804#issuecomment-2495961409).

It would be great if Tuya Local auto attempted to renew the IP address if the device goes unavailable.

This could fix situations where the router assigns a different IP to the device after initial configuration.

And would avoid the hassle of assigning static DHCP leases to every Tuya device to counteract it.

### Confirmation

- [X] I am sure this is a bug or improvement, that is well enough described that it can be implemented.
- [X] If this is for a specific device, the device config is mentioned above, or diagnostics are included.

- [X] Log messages or diagnostics relevant to the issue are included.

- [X] This is not requesting addition of a new device.

**Comment by felipecrs** [comment]
Should I just type _auto_ in the _IP address or hostname_ text box? I don't see any hint about this written anywhere.

**Comment by make-all** [comment]
Yes, the string "Auto" should trigger this feature. The advertisement of this feature was removed a few months back because advertising it seems to encourage people to use it and then report bugs related to the connection problems that the feature causes.

**Comment by t0mc1k** [comment]
Is there any way to set up automatic search? because my router doesn't support static IP address assignment and I can't set devices to static IP address (like light bulbs and other devices)
